packages/editor-ui/src/stores/ui.store.ts

```diff
@@ -87,545 +81,660 @@ try {
+	) => {
+		const { usageLeft, trialDaysLeft, userIsTrialing } = cloudPlanStore;
+		const { executionsLeft, workflowsLeft } = usageLeft;
+		const deploymentType = settingsStore.deploymentType;
+
+		telemetryStore.track('User clicked upgrade CTA', {
+			source,
+			isTrial: userIsTrialing,
+			deploymentType,
+			trialDaysLeft,
+			executionsLeft,
+			workflowsLeft,
+		});
+
+		const upgradeLink = await upgradeLinkUrl.value(source, utm_campaign, deploymentType);
+
+		if (mode === 'open') {
+			window.open(upgradeLink, '_blank');
+		} else {
+			location.href = upgradeLink;
+		}
+	};
+
+	const removeBannerFromStack = (name: BannerName) => {
+		bannerStack.value = bannerStack.value.filter((bannerName) => bannerName !== name);
+	};
+
+	const dismissBanner = async (name: BannerName, type: 'temporary' | 'permanent' = 'temporary') => {
+		if (type === 'permanent') {
+			await dismissBannerPermanently(rootStore.restApiContext, {
+				bannerName: name,
+				dismissedBanners: settingsStore.permanentlyDismissedBanners,
 			});
+			removeBannerFromStack(name);
+			return;
+		}
+		removeBannerFromStack(name);
+	};
 
-			const upgradeLink = await this.upgradeLinkUrl(source, utm_campaign, deploymentType);
+	const updateBannersHeight = (newHeight: number) => {
+		bannersHeight.value = newHeight;
+	};
 
-			if (mode === 'open') {
-				window.open(upgradeLink, '_blank');
-			} else {
-				location.href = upgradeLink;
-			}
-		},
-		async dismissBanner(
-			name: BannerName,
-			type: 'temporary' | 'permanent' = 'temporary',
-		): Promise<void> {
-			if (type === 'permanent') {
-				await dismissBannerPermanently(useRootStore().restApiContext, {
-					bannerName: name,
-					dismissedBanners: useSettingsStore().permanentlyDismissedBanners,
-				});
-				this.removeBannerFromStack(name);
-				return;
-			}
-			this.removeBannerFromStack(name);
-		},
-		updateBannersHeight(newHeight: number): void {
-			this.bannersHeight = newHeight;
-		},
-		pushBannerToStack(name: BannerName) {
-			if (this.bannerStack.includes(name)) return;
-			this.bannerStack.push(name);
-		},
-		removeBannerFromStack(name: BannerName) {
-			this.bannerStack = this.bannerStack.filter((bannerName) => bannerName !== name);
-		},
-		clearBannerStack() {
-			this.bannerStack = [];
-		},
-		getNotificationsForView(view: VIEWS): NotificationOptions[] {
-			return this.pendingNotificationsForViews[view] ?? [];
-		},
-		setNotificationsForView(view: VIEWS, notifications: NotificationOptions[]) {
-			this.pendingNotificationsForViews[view] = notifications;
-		},
-		deleteNotificationsForView(view: VIEWS) {
-			delete this.pendingNotificationsForViews[view];
-		},
-	},
+	const pushBannerToStack = (name: BannerName) => {
+		if (bannerStack.value.includes(name)) return;
+		bannerStack.value.push(name);
+	};
+
+	const clearBannerStack = () => {
+		bannerStack.value = [];
+	};
+
+	const getNotificationsForView = (view: VIEWS) => {
+		return pendingNotificationsForViews.value[view] ?? [];
+	};
```

Comment:
This function is used in only once place. Can we remove it and use `pendingNotificationsForViews` directly instead?